Patch Tuesday 🐸

paescuj opened the pull request:
- [x] Update `@directus/tsconfig` (https://github.com/directus/tsconfig/pull/5) & revert `tsconfig.json` files
- [x] Bump pnpm to latest version (8.3.1)

## `directus-monorepo`

```
 @typescript-eslint/eslint-plugin  5.55.0  →  5.59.0
 @typescript-eslint/parser         5.55.0  →  5.59.0
 eslint                            8.36.0  →  8.38.0
 eslint-config-prettier             8.7.0  →   8.8.0
 eslint-plugin-vue                  9.9.0  →  9.11.0
 prettier                           2.8.4  →   2.8.7
```

## `@directus/api`

```
 @aws-sdk/client-ses          3.292.0  →   3.316.0
 @godaddy/terminus             4.11.2  →    4.12.0
 @rollup/plugin-alias           4.0.3  →     5.0.0
 @rollup/plugin-node-resolve   15.0.1  →    15.0.2
 @types/async                  3.2.18  →    3.2.19
 @types/inquirer                8.2.6  →     9.0.3
 @types/node                  18.15.3  →  18.15.13
 @vitest/coverage-c8           0.29.3  →    0.30.1
 axios                          1.3.4  →     1.3.6
 camelcase                      6.3.0  →     7.0.1
 chalk                          4.1.2  →     5.2.0
 commander                      9.5.0  →    10.0.1
 execa                          5.1.1  →     7.1.1
 fs-extra                      11.1.0  →    11.1.1
 helmet                         6.0.1  →     6.1.5
 icc                            2.0.0  →     3.0.0
 inquirer                       8.2.5  →     9.1.5
 ioredis                        5.3.1  →     5.3.2
 joi                           17.8.4  →    17.9.1
 liquidjs                      10.6.1  →    10.7.0
 marked                        4.2.12  →     4.3.0
 nanoid                         3.3.4  →     4.0.2
 ora                            5.4.1  →     6.3.0
 rollup                        3.19.1  →    3.20.7
 sharp                         0.31.3  →    0.32.0
 strip-bom-stream               4.0.0  →     5.0.0
 tedious                       15.1.3  →    16.0.0
 typescript                     4.9.5  →     5.0.4
 vitest                        0.29.3  →    0.30.1
 vm2                           3.9.16  →    3.9.17
```

#### :x: Can't be updated at this time

- `ldapjs 2 → 3` Bigger change, requires separate PR
- `openapi3-ts 3 → 4` Bigger change, requires separate PR

## `@directus/app`

Includes update to [Vite 4.3](https://vitejs.dev/blog/announcing-vite4-3.html) 🚀 

```
 @babel/core                             7.21.3  →    7.21.4
 @babel/preset-env                       7.20.2  →    7.21.4
 @fortawesome/fontawesome-svg-core        6.3.0  →     6.4.0
 @fortawesome/free-brands-svg-icons       6.3.0  →     6.4.0
 @fullcalendar/core                       6.1.4  →     6.1.5
 @fullcalendar/daygrid                    6.1.4  →     6.1.5
 @fullcalendar/interaction                6.1.4  →     6.1.5
 @fullcalendar/list                       6.1.4  →     6.1.5
 @fullcalendar/timegrid                   6.1.4  →     6.1.5
 @pinia/testing                          0.0.15  →    0.0.16
 @popperjs/core                          2.11.6  →    2.11.7
 @storybook/addon-actions            7.0.0-rc.4  →     7.0.6
 @storybook/addon-backgrounds        7.0.0-rc.4  →     7.0.6
 @storybook/addon-docs               7.0.0-rc.4  →     7.0.6
 @storybook/addon-essentials         7.0.0-rc.4  →     7.0.6
 @storybook/addon-links              7.0.0-rc.4  →     7.0.6
 @storybook/addon-mdx-gfm            7.0.0-rc.4  →     7.0.6
 @storybook/addon-measure            7.0.0-rc.4  →     7.0.6
 @storybook/addon-outline            7.0.0-rc.4  →     7.0.6
 @storybook/client-api               7.0.0-rc.4  →     7.0.6
 @storybook/client-logger            7.0.0-rc.4  →     7.0.6
 @storybook/vue3                     7.0.0-rc.4  →     7.0.6
 @storybook/vue3-vite                7.0.0-rc.4  →     7.0.6
 @types/diff                              5.0.2  →     5.0.3
 @types/dompurify                         3.0.0  →     3.0.2
 @types/lodash                         4.14.191  →  4.14.194
 @vitejs/plugin-vue                       4.0.0  →     4.1.0
 @vue/test-utils                          2.3.1  →     2.3.2
 apexcharts                              3.37.1  →    3.39.0
 axios                                    1.3.4  →     1.3.6
 dompurify                                3.0.1  →     3.0.2
 happy-dom                                8.9.0  →     9.8.4
 marked                                  4.2.12  →     4.3.0
 nanoid                                   4.0.1  →     4.0.2
 pinia                                   2.0.33  →    2.0.35
 sass                                    1.59.3  →    1.62.0
 storybook                           7.0.0-rc.4  →     7.0.6
 typescript                               4.9.5  →     5.0.4
 vite                                     4.1.4  →     4.3.1
 vitest                                  0.29.3  →    0.30.1
 webpack                                 5.76.2  →    5.80.0
```

#### :x: Can't be updated at this time

- `codemirror 5 → 6` Looks like codemirror 6 is a total rewrite and restructure of codemirror, meaning that the migration path is effectively the same as replacing codemirror completely. This should be done as a separate effort.
- `mapbox-gl 1 → 2` https://github.com/directus/directus/issues/16185
- `maplibre-gl 1 → 2` https://github.com/directus/directus/issues/16185

## `@directus/composables`

```
 @vitest/coverage-c8  0.29.3  →  0.30.1
 @vue/test-utils       2.3.1  →   2.3.2
 axios                 1.3.4  →   1.3.6
 typescript            4.9.5  →   5.0.4
 vitest               0.29.3  →  0.30.1
```

##  `@directus/constant`

```
 typescript  4.9.5  →  5.0.4
```

##  `create-directus-extension`

```
 inquirer  8.2.5  →  9.1.5
```

##  `@directus/exceptions`

```
 @vitest/coverage-c8  0.29.3  →  0.30.1
 joi                  17.8.4  →  17.9.1
 typescript            4.9.5  →   5.0.4
 vitest               0.29.3  →  0.30.1
```

##  `@directus/extensions-sdk`

```
 @rollup/plugin-node-resolve   15.0.1  →   15.0.2
 @types/inquirer                8.2.6  →    9.0.3
 @vitest/coverage-c8           0.29.3  →   0.30.1
 chalk                          4.1.2  →    5.2.0
 commander                      9.5.0  →   10.0.1
 concurrently                   7.6.0  →    8.0.1
 esbuild                      0.17.11  →  0.17.17
 execa                          5.1.1  →    7.1.1
 fs-extra                      11.1.0  →   11.1.1
 inquirer                       8.2.5  →    9.1.5
 ora                            5.4.1  →    6.3.0
 rollup                        3.19.1  →   3.20.7
 typescript                     4.9.5  →    5.0.4
 vitest                        0.29.3  →   0.30.1
```

##  `@directus/schema`

```
 typescript  4.9.5  →  5.0.4
```

## `@directus/specs`

#### :x: Can't be updated at this time

- `openapi3-ts 3 → 4` Bigger change, requires separate PR

## `@directus/storage`

```
 @vitest/coverage-c8  0.29.3  →  0.30.1
 typescript            4.9.5  →   5.0.4
 vitest               0.29.3  →  0.30.1
```

---

## `@directus/storage-driver-azure`

```
 @azure/storage-blob  12.13.0  →  12.14.0
 @vitest/coverage-c8   0.29.3  →   0.30.1
 typescript             4.9.5  →    5.0.4
 vitest                0.29.3  →   0.30.1
```

---

## `@directus/storage-driver-cloudinary`

```
 @vitest/coverage-c8  0.29.3  →  0.30.1
 typescript            4.9.5  →   5.0.4
 undici               5.21.0  →  5.22.0
 vitest               0.29.3  →  0.30.1
```

---

## `@directus/storage-driver-gcs`

```
 @google-cloud/storage   6.9.4  →   6.9.5
 @vitest/coverage-c8    0.29.3  →  0.30.1
 typescript              4.9.5  →   5.0.4
 vitest                 0.29.3  →  0.30.1
```

---

## `@directus/storage-driver-local`

```
 @vitest/coverage-c8  0.29.3  →  0.30.1
 typescript            4.9.5  →   5.0.4
 vitest               0.29.3  →  0.30.1
```

---

## `@directus/storage-driver-s3`

```
 @aws-sdk/abort-controller  3.292.0  →  3.310.0
 @aws-sdk/client-s3         3.292.0  →  3.317.0
 @aws-sdk/lib-storage       3.292.0  →  3.317.0
 @vitest/coverage-c8         0.29.3  →   0.30.1
 typescript                   4.9.5  →    5.0.4
 vitest                      0.29.3  →   0.30.1
```

## `@directus/types`

```
 typescript  4.9.5  →  5.0.4
```

## `@directus/update-check`

```
 @vitest/coverage-c8  0.29.3  →  0.30.1
 semver                7.3.8  →   7.5.0
 typescript            4.9.5  →   5.0.4
 undici               5.21.0  →  5.22.0
 vitest               0.29.3  →  0.30.1
```

## `@directus/utils`

```
 @types/node          18.15.3  →  18.15.13
 @vitest/coverage-c8   0.29.3  →    0.30.1
 concurrently           7.6.0  →     8.0.1
 fs-extra              11.1.0  →    11.1.1
 joi                   17.8.4  →    17.9.1
 typescript             4.9.5  →     5.0.4
 vitest                0.29.3  →    0.30.1
```

## `tests-blackbox`

```
 @types/jest      29.4.4  →    29.5.1
 @types/lodash  4.14.191  →  4.14.194
 axios             1.3.4  →     1.3.6
 listr2            5.0.8  →     6.3.1
 ts-jest          29.0.5  →    29.1.0
 typescript        4.9.5  →     5.0.4
```

#### :x: Can't be updated at this time

- `globby 11 → 13` ESM only

## `docs`

```
 vitepress     1.0.0-alpha.65  →  1.0.0-alpha.73
```

Switch from `@vueuse/head` to `@unhead/vue`, see https://github.com/vueuse/head#-sunsetting-vueusehead.

paescuj committed:
Update `@directus/api` deps (minor)

 @aws-sdk/client-ses          3.292.0  →   3.316.0
 @godaddy/terminus             4.11.2  →    4.12.0
 @rollup/plugin-alias           4.0.3  →     4.0.4
 @rollup/plugin-node-resolve   15.0.1  →    15.0.2
 @types/node                  18.15.3  →  18.15.11
 @vitest/coverage-c8           0.29.3  →    0.30.1
 axios                          1.3.4  →     1.3.6
 fs-extra                      11.1.0  →    11.1.1
 helmet                         6.0.1  →     6.1.5
 ioredis                        5.3.1  →     5.3.2
 joi                           17.8.4  →    17.9.1
 liquidjs                      10.6.1  →    10.7.0
 marked                        4.2.12  →     4.3.0
 nanoid                         3.3.4  →     3.3.6
 rollup                        3.19.1  →    3.20.6
 sharp                         0.31.3  →    0.32.0
 vitest                        0.29.3  →    0.30.1
 vm2                           3.9.16  →    3.9.17

paescuj committed:
Update `@directus/api` deps (major)

 @rollup/plugin-alias     4.0.4  →     5.0.0
 @types/inquirer          8.2.6  →     9.0.3
 @types/node           18.15.11  →  18.15.12
 camelcase                6.3.0  →     7.0.1
 chalk                    4.1.2  →     5.2.0
 commander                9.5.0  →    10.0.1
 execa                    5.1.1  →     7.1.1
 icc                      2.0.0  →     3.0.0
 inquirer                 8.2.5  →     9.1.5
 ~ldapjs                   2.3.3  →     3.0.2~
 nanoid                   3.3.6  →     4.0.2
 ~openapi3-ts              3.2.0  →     4.1.1~
 ora                      5.4.1  →     6.3.0
 strip-bom-stream         4.0.0  →     5.0.0
 tedious                 15.1.3  →    16.0.0
 typescript               4.9.5  →     5.0.4

paescuj committed:
Update `@directus/app` deps

 @babel/core                             7.21.3  →    7.21.4
 @babel/preset-env                       7.20.2  →    7.21.4
 @fortawesome/fontawesome-svg-core        6.3.0  →     6.4.0
 @fortawesome/free-brands-svg-icons       6.3.0  →     6.4.0
 @fullcalendar/core                       6.1.4  →     6.1.5
 @fullcalendar/daygrid                    6.1.4  →     6.1.5
 @fullcalendar/interaction                6.1.4  →     6.1.5
 @fullcalendar/list                       6.1.4  →     6.1.5
 @fullcalendar/timegrid                   6.1.4  →     6.1.5
 @pinia/testing                          0.0.15  →    0.0.16
 @popperjs/core                          2.11.6  →    2.11.7
 @storybook/addon-actions            7.0.0-rc.4  →     7.0.6
 @storybook/addon-backgrounds        7.0.0-rc.4  →     7.0.6
 @storybook/addon-docs               7.0.0-rc.4  →     7.0.6
 @storybook/addon-essentials         7.0.0-rc.4  →     7.0.6
 @storybook/addon-links              7.0.0-rc.4  →     7.0.6
 @storybook/addon-mdx-gfm            7.0.0-rc.4  →     7.0.6
 @storybook/addon-measure            7.0.0-rc.4  →     7.0.6
 @storybook/addon-outline            7.0.0-rc.4  →     7.0.6
 @storybook/client-api               7.0.0-rc.4  →     7.0.6
 @storybook/client-logger            7.0.0-rc.4  →     7.0.6
 @storybook/vue3                     7.0.0-rc.4  →     7.0.6
 @storybook/vue3-vite                7.0.0-rc.4  →     7.0.6
 @types/diff                              5.0.2  →     5.0.3
 @types/dompurify                         3.0.0  →     3.0.2
 @types/lodash                         4.14.191  →  4.14.194
 @vitejs/plugin-vue                       4.0.0  →     4.1.0
 @vue/test-utils                          2.3.1  →     2.3.2
 apexcharts                              3.37.1  →    3.39.0
 axios                                    1.3.4  →     1.3.6
 dompurify                                3.0.1  →     3.0.2
 happy-dom                                8.9.0  →     9.8.4
 marked                                  4.2.12  →     4.3.0
 nanoid                                   4.0.1  →     4.0.2
 pinia                                   2.0.33  →    2.0.34
 sass                                    1.59.3  →    1.62.0
 storybook                           7.0.0-rc.4  →     7.0.6
 typescript                               4.9.5  →     5.0.4
 vite                                     4.1.4  →     4.2.2
 vitest                                  0.29.3  →    0.30.1
 webpack                                 5.76.2  →    5.80.0

paescuj committed:
Update root deps

paescuj committed:
Update `@directus/composables` deps

paescuj committed:
Update `@directus/constant` deps

paescuj committed:
Update `create-directus-extension` deps

paescuj committed:
Update `@directus/exceptions` deps

paescuj committed:
tsconfig workaround: ignoreDeprecations

paescuj committed:
Update `@directus/extensions-sdk` deps

paescuj committed:
Update `@directus/schema` deps

paescuj committed:
Update `@directus/storage` deps

paescuj committed:
Update `@directus/storage-driver-azure` deps

paescuj committed:
Update `@directus/storage-driver-cloudinary` deps

paescuj committed:
Update `@directus/storage-driver-gcs` deps

paescuj committed:
Update `@directus/storage-driver-local` deps

paescuj committed:
Update `@directus/storage-driver-s3` deps

paescuj committed:
Update `@directus/types` deps

paescuj committed:
Update `@directus/update-check` deps

paescuj committed:
Update `@directus/utils` deps

paescuj committed:
tsconfig workaround in schema: ignoreDeprecations

paescuj committed:
tsconfig workaround in remaining packages: ignoreDeprecations

paescuj committed:
Update `tests-blackbox` deps

rijkvanzanten committed:
Merge branch 'main' into patch-tuesday-apr-23

rijkvanzanten committed:
Revert "tsconfig workaround: ignoreDeprecations"

This reverts commit 5d97da55e3ae2083c6679d88d852fb8bf1ba10c6.

rijkvanzanten committed:
Revert tsconfig override

rijkvanzanten committed:
Update tsconfig

rijkvanzanten committed:
Fix imports in @directus/utils

rijkvanzanten committed:
Fix imports in composables

rijkvanzanten committed:
Fix imports in extensions-sdk

rijkvanzanten committed:
Fix imports in @directus/api

paescuj committed:
Move RateLimiterAbstract to types import as well

paescuj committed:
Bump pnpm to 8.3.1

azrikahar commented on packages/extensions-sdk/package.json:
```diff
@@ -40,31 +40,31 @@
-		"vitest": "0.29.3"
+		"@types/inquirer": "9.0.3",
+		"@vitest/coverage-c8": "0.30.1",
+		"concurrently": "8.0.1",
```
I believe we can remove `concurrently` now 👍

Seems like the script using concurrently was removed in https://github.com/directus/directus/pull/18014/files#diff-bf5b5d2513f9e215c8872acc7ae225035fc81ec09a5cee9ddcd1b8708c52a4f1L44

Adding a screenshot here since that PR is a tad big and the link may not load properly:

![](https://user-images.githubusercontent.com/42867097/233599381-83bb3ab8-59c9-4a91-9dfe-9b421a875335.png)

paescuj replied:
Thanks! 👍

azrikahar reviewed (approved):
Apart from the comment regarding concurrently, LGTM 👍

---

Perhaps this little tangent can/needs to be a separate PR for clarity, but since we're updating `helmet` and `nanoid`, as well as our recent move to ESM, should we "revert" the changes for `helmet` and `nanoid` in #16696?

eg. reverting `const { nanoid } = await import('nanoid');` to top level `import { nanoid } from 'nanoid';`.

paescuj committed:
Update `docs` deps

> [...] the @vueuse/head package will be deprecated. If you're setting up this package on a new project, you should use the @unhead/vue package directly [...]

paescuj committed:
Merge branch 'main' into patch-tuesday-apr-23

paescuj committed:
Remove obselete dep `concurrently` from extensions-sdk

Co-authored-by: Azri Kahar <42867097+azrikahar@users.noreply.github.com>

paescuj commented:
> Perhaps this little tangent can/needs to be a separate PR for clarity, but since we're updating `helmet` and `nanoid`, as well as our recent move to ESM, should we "revert" the changes for `helmet` and `nanoid` in #16696?
> 
> eg. reverting `const { nanoid } = await import('nanoid');` to top level `import { nanoid } from 'nanoid';`.

Good call! Let's do that separately - I'll create an issue for it 👍

paescuj committed:
New day, new updates

Also forgot to include minor updates of deps in `tests-blackbox` in previous
commit

rijkvanzanten committed:
Merge branch 'main' into patch-tuesday-apr-23

paescuj committed:
Fix `api` tests
